 INFO   Application startup complete.
2025-07-31 00:57:03,014 | INFO | cost_estimator_calc_service | Async SpannerClient initialized for database: projects/anbc-hcb-dev/instances/provider-de/databases/cost-estimator-secure
2025-07-31 00:57:03,015 | INFO | cost_estimator_calc_service | Connection pool size: 10, Max workers: 20
benefit_request benefitProductType='Medical' membershipID='5~265642286+34+44+20250101+784461+AM+39' planIdentifier='3~' serviceInfo=[ServiceInfo(serviceCodeInfo=ServiceCodeInfo(providerType=[ProviderType(code='IO')], placeOfService=[PlaceOfService(code='21')], providerSpecialty=[ProviderSpecialty(code=' ')], code='28054', type='CPT4'))]
CostEstimatorRateCriteria fields: {'providerIdentificationNumber': <class 'str'>, 'serviceCode': <class 'str'>, 'serviceType': <class 'str'>, 'serviceLocationNumber': typing.Optional[str], 'networkId': typing.Optional[str], 'placeOfService': <class 'str'>, 'zipCode': <class 'str'>, 'isOutofNetwork': <class 'bool'>, 'providerSpecialtyCode': typing.Optional[str], 'providerType': typing.Optional[str]}
rate_criteria CostEstimatorRateCriteria(providerIdentificationNumber='9394609', serviceCode='28054', serviceType='CPT4', serviceLocationNumber='10701', networkId='12395', placeOfService='21', zipCode='07083', isOutofNetwork=False, providerSpecialtyCode=' ', providerType='IO')
2025-07-31 00:57:03,019 | INFO | cost_estimator_calc_service | Executing async query: SELECT MAX(RATE) AS RATE
FROM CET_CLAIM_BASED_AMOUNTS
WHERE
    PROVIDER_IDENTIFICATION_NBR = @provideridentificationnumber
    AND NETWORK_ID = @networkid
    AND PLACE_OF_SERVICE_CD = @placeofservice
    AND SERVICE_CD = @servicecd
    AND SERVICE_TYPE_CD = @servicetype
2025-07-31 00:57:03,019 | INFO | cost_estimator_calc_service | Query params: {'servicecd': '28054', 'provideridentificationnumber': '9394609', 'placeofservice': '21', 'servicetype': 'CPT4', 'networkid': '12395', 'servicelocationnumber': '10701', 'providerbusinessgroupnbr': ''}
/Users/N133723/Library/Python/3.12/lib/python/site-packages/google/auth/_default.py:76: UserWarning: Your application has authenticated using end user credentials from Google Cloud SDK without a quota project. You might receive a "quota exceeded" or "API not enabled" error. See the following page for troubleshooting: https://cloud.google.com/docs/authentication/adc-troubleshooting/user-creds. 
  warnings.warn(_CLOUD_SDK_CREDENTIALS_WARNING)
2025-07-31 00:57:04,822 | INFO | cost_estimator_calc_service | Successfully retrieved new token
2025-07-31 00:57:05,199 | INFO | cost_estimator_calc_service | Successfully retrieved new token
2025-07-31 00:57:10,548 | INFO | cost_estimator_calc_service | Claim-based rate result: [[None]]
2025-07-31 00:57:10,549 | WARNING | cost_estimator_calc_service | Empty or malformed result
2025-07-31 00:57:10,549 | INFO | cost_estimator_calc_service | No claim-based rate found, proceeding to get provider info
2025-07-31 00:57:10,549 | INFO | cost_estimator_calc_service | Executing async query: 
                    SELECT DISTINCT PRODUCT_CD, PROVIDER_BUSINESS_GROUP_NBR
                    FROM CET_RATES
                    WHERE SERVICE_CD = @servicecd
                      AND SERVICE_TYPE_CD = @servicetype
                      AND PLACE_OF_SERVICE_CD = @placeofservice
                      AND PROVIDER_BUSINESS_GROUP_NBR = @providerbusinessgroupnbr
                
2025-07-31 00:57:10,549 | INFO | cost_estimator_calc_service | Query params: {'servicecd': '28054', 'provideridentificationnumber': '9394609', 'placeofservice': '21', 'servicetype': 'CPT4', 'networkid': '12395', 'servicelocationnumber': '10701', 'providerbusinessgroupnbr': ''}
/Users/N133723/Library/Python/3.12/lib/python/site-packages/google/auth/_default.py:76: UserWarning: Your application has authenticated using end user credentials from Google Cloud SDK without a quota project. You might receive a "quota exceeded" or "API not enabled" error. See the following page for troubleshooting: https://cloud.google.com/docs/authentication/adc-troubleshooting/user-creds. 
  warnings.warn(_CLOUD_SDK_CREDENTIALS_WARNING)
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1753941432.550045 2572162 fork_posix.cc:71] Other threads are currently calling into gRPC, skipping fork() handlers
2025-07-31 00:57:15,126 | INFO | cost_estimator_calc_service | Extraction result from CET_RATES: []
2025-07-31 00:57:15,126 | WARNING | cost_estimator_calc_service | Could not get provider info from CET_RATES, falling back to _get_provider_info: CET_RATES did not return any product/provider info
2025-07-31 00:57:15,127 | INFO | cost_estimator_calc_service | added providerspecialtycode
2025-07-31 00:57:15,127 | INFO | cost_estimator_calc_service | added providertype param
2025-07-31 00:57:15,127 | INFO | cost_estimator_calc_service | Executing async query: SELECT DISTINCT PROVIDER_BUSINESS_GROUP_NBR, PRODUCT_CD, RATING_SYSTEM_CD, EPDB_GEOGRAPHIC_AREA_CD
FROM CET_PROVIDERS
WHERE
    PROVIDER_IDENTIFICATION_NBR = @provideridentificationnumber
    AND NETWORK_ID = @networkid
    AND SERVICE_LOCATION_NBR = @servicelocationnumber
        AND SPECIALTY_CD = @providerspecialtycode
        AND PROVIDER_TYPE_CD = @providertype
2025-07-31 00:57:15,127 | INFO | cost_estimator_calc_service | Query params: {'servicecd': '28054', 'provideridentificationnumber': '9394609', 'placeofservice': '21', 'servicetype': 'CPT4', 'networkid': '12395', 'servicelocationnumber': '10701', 'providerbusinessgroupnbr': '', 'providerspecialtycode': ' ', 'providertype': 'IO'}
/Users/N133723/Library/Python/3.12/lib/python/site-packages/google/auth/_default.py:76: UserWarning: Your application has authenticated using end user credentials from Google Cloud SDK without a quota project. You might receive a "quota exceeded" or "API not enabled" error. See the following page for troubleshooting: https://cloud.google.com/docs/authentication/adc-troubleshooting/user-creds. 
  warnings.warn(_CLOUD_SDK_CREDENTIALS_WARNING)
I0000 00:00:1753941435.134762 2572162 fork_posix.cc:71] Other threads are currently calling into gRPC, skipping fork() handlers
2025-07-31 00:57:18,684 | INFO | cost_estimator_calc_service | Provider info result: []
2025-07-31 00:57:18,684 | INFO | cost_estimator_calc_service | No provider info results found
2025-07-31 00:57:18,684 | ERROR | cost_estimator_calc_service | Error in get_rate: float() argument must be a string or a real number, not 'NoneType'
      INFO   127.0.0.1:61676 - "POST              
             /costestimator/v1/rate HTTP/1.1" 200
